src/main.nr

```diff
@@ -7,9 +9,19 @@ fn main(value: u32, hash_value: pub u32){
 }
 
 fn calculate_hash(num: u32) -> u32 {
-    
+    let size = 4294967296;
+    let mut hash_value : u64 = 0x811C9DC5;
+    let bytes = u2b::u32_to_u8(num);
+
+    for byte in bytes {
+        let product = (hash_value * PRIME) % size;
```

Comment:
You can remove the % operation from this line by having hash_value type to u32 since Noir would take care of wrapping it in size by doing %